### ISSUE TITLE
Advanced context menu item source

### ISSUE THREAD

**Issue by art0007i** [Yellow-Dog-Man/Resonite-Issues#3341]
### Is your feature request related to a problem? Please describe.

I'd like to have a context menu item which can accept a reference using referencegrabreceiver, but i cannot do it. also for building custom arc visuals, like I have an item that works like a slider and it has a visual that fills up to the point that you slide it. currently this tool uses terribleness to work but it could be all fixed with this

### Describe the solution you'd like

basically context menu item source which can have a template slot that will be placed in the users context menu. this would allow making better context menu interactions.

### Describe alternatives you've considered

manually injecting stuff onto context menus but this is jank and annoying and not perfectly reliable

**Comment by shiftyscales** [comment]
> also for building custom arc visuals, like I have an item that works like a slider and it has a visual that fills up to the point that you slide it. currently this tool uses terribleness to work but it could be all fixed with this

Sliders are covered by #718. I think it would be best to focus your request on these kind of building blocks / desired native interactions rather than having slightly more official/supported jank / injected functionality.

Aside from sliders, did you have any other forms of interaction / functionality you would want to see as native context menu functionality that isn't currently possible? Aside from sliders, what do you commonly want to inject?

**Comment by JackTheFoxOtter** [comment]
The one mentioned above was grab receivers. I'd personally like to see this supported for context menu items as well! I've had situations in the past where I wanted to create a simple system to assign a value / reference through the context menu via drag & drop, but couldn't. An example of this would be easily assigning a custom texture to use for an avatar through the context menu.

I'd also personally prefer more default UI building blocks instead of requiring creators to implement those themselves and only providing them a simple interface to inject a custom button.

**Comment by art0007i** [comment]
well another one would be the picture of a hue wheel on a arc that is supposed to select color (it's in a submenu that has just that 1 color arc)

i personally believe is that you can never cover all edge cases with primitives. more primitives can be nice but sometimes you want something more/something different

having access to the arc would also not be jank in most cases. all of the examples I've mentioned would be very clean in my opinion.
